plugin.js

```diff
@@ -36,100 +38,127 @@ CKEDITOR.plugins.add('uploadcare', {
-      exec : function() {
+      exec: function() {
         if (typeof uploadcare == 'undefined') {
           return; // not loaded yet
         }
 
         uploadcare.plugin(function(uc) {
           var settings, element, file;
 
-          if (element = searchSelectedElement(editor, 'img')) {
+          if ((element = searchSelectedElement(editor, 'img'))) {
             file = element.getAttribute('src');
-          } else if (element = searchSelectedElement(editor, 'a')) {
+          } else if ((element = searchSelectedElement(editor, 'a'))) {
             file = element.getAttribute('href');
           }
 
           if (file && uc.utils.splitCdnUrl(file)) {
             settings = uc.settings.build(
-              uc.jQuery.extend({}, config, {multiple: false})
+              uc.jQuery.extend({}, config, { multiple: false })
             );
             file = uploadcare.fileFrom('uploaded', file, settings);
           } else {
-            settings = uc.settings.build(config)
+            settings = uc.settings.build(config);
             file = null;
           }
 
-          var dialog = uploadcare.openDialog(file, settings).done(function(selected) {
-            var files = settings.multiple ? selected.files() : [selected];
-            uc.jQuery.when.apply(null, files).done(function() {
-              uc.jQuery.each(arguments, function() {
-                console.log("this is ....", this);
-                origImg = this;
-                var imageUrl = this.cdnUrl;
-                if (this.isImage && ! this.cdnUrlModifiers) {
-                  imageUrl += '-/preview/';
-                };
+          var dialog = uploadcare
+            .openDialog(file, settings)
+            .done(function(selected) {
+              var files = settings.multiple ? selected.files() : [selected];
+              uc.jQuery.when.apply(null, files).done(function() {
+                uc.jQuery.each(arguments, function() {
+                  origImg = this;
+                  var imageUrl = this.cdnUrl;
+                  if (this.isImage && !this.cdnUrlModifiers) {
+                    imageUrl += '-/preview/';
+                  }
 
-                var successHandler = function(res) {
-                  imageUrl = res.url
+                  var successHandler = function(res) {
+                    imageUrl = res.url;
                     if (element) {
                       var widget;
-                      if (editor.widgets && (widget = editor.widgets.selected[0]) && widget.element === element) {
-                        widget.setData('src', imageUrl).setData('height', null)
+                      if (
+                        editor.widgets &&
+                        (widget = editor.widgets.selected[0]) &&
+                        widget.element === element
+                      ) {
+                        widget.setData('src', imageUrl).setData('height', null);
+                        editor.extractSelectedHtml();
```

Comment:
Lines 96 and 97 is where the change was made. I know this is an expensive-ish operation, but it was the only way I found to make sure the html was generated properly. This also seems like a bit of a rare use case, so it should be fine